app/helpers/address_helper.rb

```diff
@@ -123,4 +123,19 @@ def compare(ordnance_address, edited_address)
 
     sanitized_edited_address & sanitized_ordnance_address == sanitized_ordnance_address
   end
+
+  def remove_selected_address_changes(selected_address)
+    address_hash = JSON.parse(selected_address).to_h
+    selected_address_keys = address_hash.keys
+
+    address_to_return = address_hash.select do |key, value|
+      WANTED_VALUES.include?(key) && value.instance_of?(String)
+    end
+
+    unless address_to_return.keys == selected_address_keys
+      address_to_return.delete("UPRN")
+    end
+
+    address_to_return
```

Comment:
I think validating that we're receiving the data that we expect is a good idea 👍 

This method currently wouldn't prevent users from submitting invalid values to be stored in the session. We would need to validate the data submitted against the options we had provided. I.e. cache the valid options received from the API, and then compare the user provided data with the cached options.

Reply:
Thanks @bilbof - perhaps we can have a brief catch up on this?

Reply:
I think it's not something we'd need to do in this PR, so we can do it later if needed 👍